sa/storage-authority_test.go

```diff
@@ -245,51 +244,64 @@ func TestGetLatestValidAuthorizationMultiple(t *testing.T) {
 func TestAddCertificate(t *testing.T) {
 	sa := initSA(t)
 
-	// An example cert taken from EFF's website
-	certDER, err := ioutil.ReadFile("www.eff.org.der")
-	test.AssertNotError(t, err, "Couldn't read example cert DER")
-
-	digest, err := sa.AddCertificate(certDER, 1)
-	test.AssertNotError(t, err, "Couldn't add www.eff.org.der")
-	test.AssertEquals(t, digest, "qWoItDZmR4P9eFbeYgXXP3SR4ApnkQj8x4LsB_ORKBo")
-
-	// Example cert serial is 0x21bd4, so a prefix of all zeroes should fetch it.
-	retrievedCert, err := sa.GetCertificateByShortSerial("0000000000000000")
-	test.AssertNotError(t, err, "Couldn't get www.eff.org.der by short serial")
-	test.AssertByteEquals(t, certDER, retrievedCert.DER)
-
-	retrievedCert, err = sa.GetCertificate("00000000000000000000000000021bd4")
-	test.AssertNotError(t, err, "Couldn't get www.eff.org.der by full serial")
-	test.AssertByteEquals(t, certDER, retrievedCert.DER)
-
-	certificateStatus, err := sa.GetCertificateStatus("00000000000000000000000000021bd4")
-	test.AssertNotError(t, err, "Couldn't get status for www.eff.org.der")
+	// Create a new pending certificate
+	pendingCert := core.Certificate{RegistrationID: 1}
+	pendingCert, err := sa.NewPendingCertificate(pendingCert)
+	test.AssertNotError(t, err, "Couldn't create a new pending cert")
+	test.AssertEquals(t, pendingCert.RegistrationID, int64(1))
+	test.AssertEquals(t, pendingCert.Status, core.StatusPending)
+	test.AssertNotEquals(t, pendingCert.RequestID, "")
+	test.Assert(t, core.LooksLikeAToken(pendingCert.RequestID), "Bad request ID in pending cert")
+
+	// Try to retrieve the pending certificate
+	retrieved, err := sa.GetCertificateByRequestID(pendingCert.RequestID)
+	test.AssertNotError(t, err, "Couldn't retrieve pending certificate")
+	test.AssertEquals(t, retrieved.RequestID, pendingCert.RequestID)
+	test.AssertEquals(t, retrieved.Status, pendingCert.Status)
+
+	// Try to finalize a certificate with non-final status
+	err = sa.FinalizeCertificate(pendingCert)
+	test.AssertError(t, err, "Finalized a cert that was still pending")
+
+	// Try to finalize a certificate for which there is no pending cert
+	pendingCert.Status = core.StatusInvalid
+	err = sa.FinalizeCertificate(pendingCert)
+	test.AssertNotError(t, err, "Failed to finalize on a valid request")
```

Comment:
Here's where you should verify that the pending cert got deleted from the pending table.

```
test.Assert(t, !existingPendingCert(pendingCert.RequestID), "Failed to delete pending cert")
```